src/IceRpc.Protobuf.Generators/Internal/Parser.cs

```diff
@@ -190,7 +190,12 @@ private IReadOnlyList<ServiceMethod> GetServiceMethods(INamedTypeSymbol interfac
                 items.Length == 1,
                 "Unexpected number of arguments in attribute constructor.");
             string operationName = (string)items[0].Value!;
-            serviceMethods.Add(new ServiceMethod(dispatchMethodName: GetFullName(method), operationName));
+            serviceMethods.Add(
+                new ServiceMethod(
+                    operationName,
+                    interfaceName: $"global::{GetFullName(interfaceSymbol)}",
+                    methodName: method.Name,
+                    inputTypeName: $"global::{GetFullName(method.Parameters[0].Type)}"));
```

Comment:
This is going to result in some over-qualification if the generated service is in the same namespace. We can look into this in a separate PR

Reply:
You're right, let's fix this in another PR. This overqualification isn't critical since it's pretty hard to see the file generated by the source generator.